### ISSUE TITLE
Hard Crash at Combat Initiation

### ISSUE THREAD

**Issue by Boomermav** [CaiClone/GCDTracker#10]
Running into an infrequent issue while DRK tanking inside Temple of Qarn & Cutter's Cry.  Roughly once every two dungeon runs or so, when initiating combat of a group of enemies, the game will hard crash at the moment of agro and prompt to reboot or reboot with add-ons disabled.  Also occurred once while running AST.  

Haven't been able to consistently replicate the issue yet.

**Comment by CaiClone** [comment]
Thank you for the report.
I'm also having some trouble replicating this issue by tanking with DRK, if it happens to you or anybody else I would appreciate if you send the log my way so I can debug it.

**Comment by Boomermav** [comment]
Pretty new to all of this.  Is this the log info you were looking for?
(https://github.com/CaiClone/GCDTracker/files/10329151/12-21-2022_GCDTracker_Crash.txt)

**Comment by Boomermav** [comment]
Follow up to my previous issue.  Just ran into the same crash error.  However this time I was desynthing some random equipment in New Gridania.  Only say what was happening out the corner of my eye, but it looked like the "clip" red bar was timing out when it crashed.
(https://github.com/CaiClone/GCDTracker/files/10421348/01-15-2023_GCDTracker_Crash.txt)

**Comment by CaiClone** [comment]
Thanks for the helpful hint! I think I understand the issue now. 
Would it be possible for you to check if you have this option enabled? 
![image](https://user-images.githubusercontent.com/10381832/212786813-ca93842f-0652-40d9-801d-b27d9b4f6176.png)
Leaving it enabled could cause the wheel to display in unexpected situations, like with crafter classes. This can potentially lead to bad data being sent to the visualization library and causing the crash.
I'll work on adding some protections against this, but in the meantime, could you try disabling that option and let me know if the problem persists?

**Comment by Boomermav** [comment]
That option was disabled in my settings.  I wasn't specifically on a crafter, but just desynthing some gear laying in my bag.

![move-resize window](https://user-images.githubusercontent.com/65128752/213324568-0ba54948-8376-4a46-a48f-a763639ce88b.png)

**Comment by CaiClone** [comment]
My bad, sorry for the assumption, I forgot you don't need to be a crafter to desynthesize objects. I have some good news though, I was able to reproduce the same exception in the same line by intentionally drawing a circle with too many segments. I am not entirely sure why that could happen, but just in case that's the source of the error I have released a new version (1.0.2.6) with some safeguard against that.

**Comment by Boomermav** [comment]
Awesome!  I'll reinstall and do a bit of testing it out.  Thanks so much!!

If as bit of context could help, I've simplified the look a bit by dropping the opacity of "GCD Bar Color" & "Clipping Color" to 80, and dropped the rest to 0.  I was looking for a simple indicator I look *through* while I fight.  So I've got the ring blown up to about a 5 inch diameter in the middle of my screen.  Not sure if making it so big might cause issues or not.  But hey, data is data.
